### PR TITLE
Supervisor search presets

### PR DESCRIPTION
**Pull request by tylerpar99**
Summary:
Created buttons for potential preset options on the supervisor portal. When a preset option is clicked we create a dictionary based on the presets filtering. With these changes, there are also two new options in the `getDatatableData()` function it will now allow requests for the current user and current term filtering as well as limit non-admins to only information within their department. We have also patched a bug that would attempt to check Tracy first before checking our DB we now have it check our DB first then Tracy and if that fails we pass on the record.

To Test: 
Try each preset and ensure the result match as expected. Also if you click pending forms or current term and click through each page you should see that no result throws an error.

### CODE REVIEW

**Comment by tylerpar99** [comment]
Preset Notes:

- Current Students (Not Current Term) Current students will be anyone who has an approved form for the current AY and is attached to the current user.
- Upcoming students (Not Pending Forms), will be students who are attached to the professor and have a pending form that hasn't been approved. 
- Stay away from the department as a preset. All presets should be attached to the user.

File: app/templates/main/supervisorPortal.html [hunk 1]
```diff
@@ -23,10 +23,18 @@ <h1>Supervisor Portal</h1>
...
     <button class="btn btn-success" type="submit" id="download">Download</button>
   </form>
-    </div>
-    <br>
+</div>
+<br>
+<div class="row col-md-12" align="center">
+  <button class="btn btn-primary" type="button" id="mySupervisees">Current Students</button>
+  <button class="btn btn-primary" type="button" id="evaluationsMissing">Evaluations Missing</button>
+  <button class="btn btn-primary" type="button" id="superviseesPendingForms">Upcomings Students</button>
```
Comment (AndersonStettner) [inline app/templates/main/supervisorPortal.html]: Change to "Pending Forms"

**Review by AndersonStettner — CHANGES_REQUESTED** [review]

File: tests/code/test_formSearch.py [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+            runGetDatatableData = supervisorPortal()
+            print(runGetDatatableData)
+        with app.test_request_context(
+        "/", method="POST", data=studentIDDict):
+            runGetDatatableData = supervisorPortal()
+            print(runGetDatatableData)
+        with app.test_request_context(
+        "/", method="POST", data=allEvalDict):
+            runGetDatatableData = supervisorPortal()
+            print(runGetDatatableData)
```
Comment (AndersonStettner) [inline tests/code/test_formSearch.py]: Remove print statements

**Review by AndersonStettner — CHANGES_REQUESTED** [review]

File: app/controllers/main_routes/main_routes.py [hunk 3]
```diff
@@ -273,13 +292,30 @@ def getFormattedData(filteredSearchResults):
...
-        # TODO Skipping adding to the table. Requires database work to get SLE out from form (formHistory, to be precise)
+        # TODO: Skipping adding to the table. Requires database work to get SLE out from form (formHistory, to be precise)
 
         formattedData.append(record)
 
     return formattedData
 
+@main_bp.route('/supervisorPortal/addUserToDept', methods=['GET', 'POST'])
+def addUserToDept():
+    print("entered function")
```
Comment (AndersonStettner) [inline app/controllers/main_routes/main_routes.py]: Remove Print

File: app/controllers/main_routes/main_routes.py [hunk 4]
```diff
@@ -273,13 +292,30 @@ def getFormattedData(filteredSearchResults):
...
+    userDeptData = request.form
+    supervisorDeptRecord = SupervisorDepartment.get_or_none(supervisor = userDeptData['supervisor'], department = userDeptData['department'])
+    try:
+        if supervisorDeptRecord:
+            print("true")
```
Comment (AndersonStettner) [inline app/controllers/main_routes/main_routes.py]: Remove

File: app/controllers/main_routes/main_routes.py [hunk 5]
```diff
@@ -273,13 +292,30 @@ def getFormattedData(filteredSearchResults):
...
+            return "False"
 
+        else:
+            print("false")
```
Comment (AndersonStettner) [inline app/controllers/main_routes/main_routes.py]: Remove

File: app/controllers/main_routes/main_routes.py [hunk 6]
```diff
@@ -273,13 +292,30 @@ def getFormattedData(filteredSearchResults):
...
+            createSupervisorDeptRecord = SupervisorDepartment.create(supervisor=userDeptData['supervisor'], department=userDeptData['department'])
+            return "True"
+            print("printing somethintg")
```
Comment (AndersonStettner) [inline app/controllers/main_routes/main_routes.py]: Remove

File: app/controllers/main_routes/main_routes.py [hunk 7]
```diff
@@ -273,13 +292,30 @@ def getFormattedData(filteredSearchResults):
...
+    except:
+        print("error")
```
Comment (AndersonStettner) [inline app/controllers/main_routes/main_routes.py]: Remove print, should we be catching a specific exception?

**Review by AndersonStettner — DISMISSED** [review]

**Review by BrianRamsay — APPROVED** [review]

**Pull request merged**